pages/instructorQuestion/instructorQuestion.sql

```diff
@@ -19,12 +19,106 @@ WHERE
+        aq.question_id,
+        histogram(number_attempts, 1, 10, 10) as num_attempts_histogram,
+        a.type AS assessment_type
+    FROM
+        instance_questions AS iq
+        JOIN assessment_questions AS aq ON iq.assessment_question_id=aq.id
+        JOIN assessment_instances AS ai ON (iq.assessment_instance_id = ai.id)
+        JOIN assessments AS a ON aq.assessment_id=a.id
+        JOIN course_instances AS ci ON (a.course_instance_id = ci.id)
+        JOIN enrollments AS e ON (ai.user_id = e.user_id AND ci.id = e.course_instance_id)
+    WHERE
+        number_attempts != 0
+        AND aq.deleted_at IS NULL
+        AND e.role = 'Student'
+    GROUP BY
+        aq.question_id,
+        a.type
+),
+num_attempts_gave_up_histogram AS (
+    SELECT
+        aq.question_id,
+        histogram(number_attempts, 1, 10, 10) as num_attempts_histogram,
+        a.type AS assessment_type
+    FROM
+        instance_questions AS iq
+        JOIN assessment_questions AS aq ON iq.assessment_question_id=aq.id
+        JOIN assessment_instances AS ai ON (iq.assessment_instance_id = ai.id)
+        JOIN assessments AS a ON (ai.assessment_id = a.id)
+        JOIN course_instances AS ci ON (a.course_instance_id = ci.id)
+        JOIN enrollments AS e ON (ai.user_id = e.user_id AND ci.id = e.course_instance_id)
+    WHERE
+        number_attempts != 0
+        AND iq.points = 0
+        AND aq.deleted_at IS NULL
+        AND e.role = 'Student'
+    GROUP BY
+        aq.question_id,
+        a.type
+)
 SELECT
     q.*,
     to_json(top) AS topic,
     tags_for_question(q.id) AS tags,
-    assessments_for_question(q.id, $2) AS assessments
+    assessments_for_question(q.id, $2) AS assessments,
+    num_attempts_histogram.num_attempts_histogram AS question_attempts_histogram,
+    num_attempts_gave_up_histogram.num_attempts_histogram AS question_attempts_before_giving_up_histogram,
+    num_attempts_histogram_hw.num_attempts_histogram AS question_attempts_histogram_hw,
+    num_attempts_gave_up_histogram_hw.num_attempts_histogram AS question_attempts_before_giving_up_histogram_hw
+
 FROM questions as q
 JOIN topics as top ON (top.id = q.topic_id)
-AND q.id = $1
-AND q.deleted_at IS NULL;
+LEFT JOIN num_attempts_histogram ON (num_attempts_histogram.question_id = q.id AND
+num_attempts_histogram.assessment_type = 'Exam')
+LEFT JOIN num_attempts_gave_up_histogram ON (num_attempts_gave_up_histogram.question_id = q.id AND
+num_attempts_gave_up_histogram.assessment_type = 'Exam')
+LEFT JOIN num_attempts_histogram AS num_attempts_histogram_hw
+        ON (num_attempts_histogram_hw.question_id = q.id AND num_attempts_histogram_hw.assessment_type = 'Homework')
+LEFT JOIN num_attempts_gave_up_histogram AS num_attempts_gave_up_histogram_hw
+        ON (num_attempts_gave_up_histogram_hw.question_id = q.id AND
+            num_attempts_gave_up_histogram_hw.assessment_type = 'Homework')
+WHERE
+    q.id = $1
+    AND q.deleted_at IS NULL;
+
+-- BLOCK assessment_stats
```

Comment:
Well, the statistics are mostly used in the "Detailed assessment statistics" block within the instructor question page, so I think the best name is assessment_question_stats.